sirepo/package_data/static/js/sirepo.js

```diff
@@ -2147,4 +2225,47 @@ SIREPO.app.controller('SimulationsController', function (appState, panelState, r
         appState.saveQuietly('simulationFolder');
     });
     loadList();
+
+    // invoked in loadList() callback
+    function checkURLForFolder() {
+
+        var pathNoSectionRE = new RegExp('^\/' + activeSection.getActiveSection());
+        var decodedPath = decodeURI($location.path());
+        var canonicalPath = compoundPathToPath(decodedPath.replace(pathNoSectionRE, ''));
+
+        if( canonicalPath !== (appState.getActiveFolderPath() || '/') ) {
+            var fList = [rootFolder()];
+            var newFolder = folderForPathInList(canonicalPath, fList);
+            if( newFolder ) {
+                appState.setActiveFolderPath(canonicalPath);
+                appState.setActiveFolder(newFolder);
+                self.openItem(newFolder);
+            }
+            else {
+                requestSender.localRedirect('notFound');
+            }
+        }
+        else {
+            if( !appState.getActiveFolder() ) {
+                self.openItem(rootFolder());
+            }
+        }
+    }
+    function folderForPathInList (folderPath, folderList) {
+        for( var i = 0; i < folderList.length; ++i ) {
+            var item = folderList[i];
+            if( item.isFolder ) {
+                var lPath = self.pathName(item);
```

Comment:
I'm confused about compoundPath vs pathName. Seems like they are similar.

Reply:
They are identical except for the delimiter. I guess pathName is just used to depict an artificial file system on the UI so we could change its delimiter and eliminate compoundPath - but I'd want to confirm that's OK!